plugins/plugin-chart-echarts/src/Graph/transformProps.ts

```diff
@@ -0,0 +1,232 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  getMetricLabel,
+  DataRecord,
+  DataRecordValue,
+} from '@superset-ui/core';
+import { EChartsOption, GraphSeriesOption } from 'echarts';
+import { extent as d3Extent } from 'd3-array';
+import { GraphEdgeItemOption } from 'echarts/types/src/chart/graph/GraphSeries';
+import {
+  EchartsGraphFormData,
+  EChartGraphNode,
+  DEFAULT_FORM_DATA as DEFAULT_GRAPH_FORM_DATA,
+} from './types';
+import { DEFAULT_GRAPH_SERIES_OPTION, NORMALIZATION_LIMITS } from './constants';
+import { EchartsProps } from '../types';
+import { getChartPadding, getLegendProps } from '../utils/series';
+
+/**
+ * Normalize node size, edge width, and apply label visibility thresholds.
+ */
+function normalizeStyles(
+  nodes: EChartGraphNode[],
+  links: GraphEdgeItemOption[],
+  {
+    showSymbolThreshold,
+  }: {
+    showSymbolThreshold?: number;
+  },
+) {
+  const [minValue, maxValue] = d3Extent(nodes, x => x.value) as [number, number];
+  const spread = maxValue - minValue;
+  nodes.forEach(node => {
+    // eslint-disable-next-line no-param-reassign
+    node.symbolSize =
+      (((node.value - minValue) / spread) * NORMALIZATION_LIMITS.maxNodeSize || 0) +
+      NORMALIZATION_LIMITS.minNodeSize;
+    // eslint-disable-next-line no-param-reassign
+    node.label = {
+      ...node.label,
+      show: showSymbolThreshold ? node.value > showSymbolThreshold : true,
+    };
+  });
+  links.forEach(link => {
```

Comment:
Link width goes -Infinity if metric is null,does it need to be handled?

Reply:
yes please

Reply:
shouldn't we calculate min,max,spread for links again?
Because if values of nodes used to normalize links,it gives negative width and it disappears from graph.
@ktmud

Reply:
added.